### ISSUE TITLE
PR Discussion - Traitlet config passthrough

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#1396]
# Traitlet config passthrough
In BinderHub, @minrk has made it so that various configuration, like c.KubeSpawner.some_configuration can be configured directly through the provided Helm values without the need for custom steps! It allows us to avoid having to explicitly provide a value in the Helm chart that we explicitly loads for use in jupyterhub_config.py! We should do this for z2jh's JupyterHub as well as it makes this whole deployment a lot more sustainable to maintain and less error prone going on.

## Reference solution in BinderHub
In binderhub_config.py for the binder pod of a BinderHub, which is like this chart's hub pod's jupyterhub_config.py, we [find the following](https://github.com/jupyterhub/binderhub/blob/3fe81437bc734458e4c2c61a29a6612533839e3c/helm-chart/images/binderhub/binderhub_config.py#L65-L67):

```python
# load config from values.yaml
for section, sub_cfg in get_value('config', {}).items():
    c[section].update(sub_cfg)
```

... and [in BinderHub's values.yaml](https://github.com/jupyterhub/binderhub/blob/3fe81437bc734458e4c2c61a29a6612533839e3c/helm-chart/binderhub/values.yaml#L37-L38) we find:

```yaml
config:
  BinderHub: {}
```

## PR work to be done
- [x] Await #1407 merge
- [ ] Create a blank entry in values.yaml and write a comment about it, preferably also with an example of how it can be used, for example [similarly to this](https://gitlab.com/gitlab-org/charts/gitlab/blob/master/values.yaml#L204-217).
- [x] Load the configuration in jupyterhub_config.py, just like in BinderHub.
- [ ] Add documentation in schema.yaml
- [ ] Perhaps add a deprecation logic, or create an issue to catch some old ways of doing this

**Comment by consideRatio** [comment]
I think this is very, very, relevant, for example to resolve the chaos of configuring the Authenticators.

**Comment by consideRatio** [comment]
Closed along with #1871. We have it under hub.config now, what follows is the need to avoid using two sets of configurations (hub.baseUrl and hub.config.base_url).
